<launch>

	<arg name="imu_frame" default="imu_frame"/>
	<arg name="base_frame" default="base_frame"/>
	<arg name="cam_name" default="cam_name"/>
	
	<arg name="depth_topic" default="depth_in"/>
	<arg name="cam_info_topic" default="cam_info_in"/>
	
	<arg name="viz_topic" default="viz_out"/>
	<arg name="dom_pt_topic" default="dom_pt_out"/>
	
	<arg name="launch_noise_cloud" default="false"/>
	
	<! .................................................... -->

	<node ns="$(arg cam_name)" name="filter_point_node" pkg="local_planning" type="filter_point_node" output="screen">
		
		<remap from="filter_point_node/pt_cloud_in" to="$(arg depth_topic)"/>
		<remap from="filter_point_node/cam_info_in" to="$(arg cam_info_topic)"/>
		
		<remap from="filter_point_node/pt_cloud_viz" to="$(arg viz_topic)"/>
		<remap from="filter_point_node/pt_out" to="$(arg dom_pt_topic)"/>
			
		<param name="distance_interval" type="double" value="0.1"/>
		<param name="pixel_interval" type="int" value="50"/>
		<param name="min_distance" type="double" value="0.15"/>
		<param name="max_distance" type="double" value="1.2"/>
		<param name="n_particles_vox" type="int" value="50000"/>
		<param name="outliers_per_voxel" type="int" value="20"/>
			
		<rosparam param="trans_noise_sdev"> [8, 0.4] </rosparam> <!-- voxels, distance -->
		<rosparam param="obsv_noise_sdev"> [200, 10] </rosparam> <!-- confidence points, nOutliers -->
			
		<rosparam param="max_action_values"> [0.0, 0.4, 0.4] </rosparam>
		<rosparam param="min_action_values"> [-0.4, -0.4, -0.4] </rosparam>
		<rosparam param="action_intervals"> [0.1, 0.1, 0.1] </rosparam>
			
		<rosparam param="reward_Q"> [0.7, 0.7, 0.7] </rosparam>
		<param name="repulsive_potential_max_distance" type="double" value="1.2"/>
		<param name="repulsive_potential_gain" type="double" value="2"/>
			
		<param name="alpha_vector_iterations" type="int" value="10"/>
			
		<param name="lookahead_time" type="double" value="0.5"/>
		<param name="sampling_time" type="double" value="0.1"/>
		<param name="base_frame_id" type="string" value="$(arg base_frame)"/>
			
	</node>

	<! .................................................... -->

	<node if="$(arg launch_noise_cloud)" name="noise_cloud_node" pkg="local_planning" type="noise_cloud_node" output="screen">
		
		<remap from="noise_cloud_node/pt_cloud_in" to="/royale_camera_driver/point_cloud"/>
			
	</node>

	<! .................................................... -->
	
</launch>
